parser.hh

```diff
@@ -199,9 +206,15 @@ class Input
 	 * so that the first request will trigger a fetch from the underlying
 	 * storage.
 	 */
-	Input() : buffer(0), buffer_start(1), buffer_end(0) {}
+	Input(std::string name)
```

Comment:
Well, look at that: I thought there was elision of such things, but apparently there isn't. I blame Sutter and Alexandrescu.